openedx/features/edly/tests/test_middleware.py

```diff
@@ -71,3 +74,91 @@ def test_super_user_has_all_sites_access(self):
 
         response = client.get('/', follow=True)
         assert response.status_code == 200
+
+
+class SettingsOverrideMiddlewareTests(TestCase):
+    """
+    Tests for "SettingsOverrideMiddleware".
+    """
+    def setUp(self):
+        """
+        Set up values for tests.
+        """
+        self.user = EdlyUserFactory()
+        self.request = RequestFactory().get('/')
+        self.request.user = self.user
+        self.request.site = SiteFactory()
+        EdlySubOrganizationFactory(lms_site=self.request.site)
+
+        self.client = Client(SERVER_NAME=self.request.site.domain)
+        self.client.cookies.load(
+            {
+                settings.EDLY_USER_INFO_COOKIE_NAME: cookies._get_edly_user_info_cookie_string(self.request)
+            }
+        )
+        self.client.login(username=self.user.username, password='test')
+
+    def test_settings_override_middleware_logs_warning_if_no_site_configuration_is_present(self):
+        """
+        Tests "SettingsOverrideMiddleware" logs warning if no site configuration is present.
+        """
+        with LogCapture(LOGGER_NAME) as logger:
+            self.client.get('/', follow=True)
+            logger.check_present(
+                (
+                    LOGGER_NAME,
+                    'WARNING',
+                    'Site ({site}) has no related site configuration.'.format(site=self.request.site)
+                )
+            )
+
+    def test_settings_override_middleware_logs_warning_if_site_configuration_has_no_values_set(self):
+        """
+        Tests "SettingsOverrideMiddleware" logs warning if site configuration has no values.
+        """
+        SiteConfigurationFactory(site=self.request.site)
+        with LogCapture(LOGGER_NAME) as logger:
+            self.client.get('/', follow=True)
+            logger.check_present(
+                (
+                    LOGGER_NAME,
+                    'WARNING',
+                    'Site configuration for site ({site}) has no values set.'.format(site=self.request.site)
+                )
+            )
+
+    def test_settings_override_middleware_logs_warning_if_site_configuration_has_no_django_settings_overrides(self):
```

Comment:
nit: we should avoid such long names for test methods

Reply:
just use `test_settings_override_middleware_logs_warning`